File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    q = int(a%2 != 0)
    w = int(b%2 != 0)
    
    if q+w == 2:
        return a**2 + b**2
    elif q+w == 1:
        return 2*(a+b)
    else:
        return a-b if a>b else b-a
        

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        if ( a % 2 == 1 && b % 2 == 1) {
            return (int) Math.pow(a, 2) + (int) Math.pow(b, 2);
        } else if ( a % 2 == 0 && b % 2 == 0) {
            return Math.abs(a - b);
        }
        return (a + b) * 2;
    }
}